js/modules/loki_app_dot_net_api.js

```diff
@@ -800,104 +838,222 @@ class LokiPublicChannelAPI {
+            // why am I getting these?
+            if (messageData === undefined) {
+              log.warn('invalid pendingMessages');
+              return;
+            }
+            // if a known slave, queue
+            if (slavePrimaryMap[messageData.source]) {
+              // delay sending the message
+              if (slaveMessages[messageData.source] === undefined) {
+                slaveMessages[messageData.source] = [messageData];
+              } else {
+                slaveMessages[messageData.source].push(messageData);
+              }
+            } else {
+              // no user or isPrimary means not multidevice, send event now
+              this.serverAPI.chatAPI.emit('publicMessage', {
+                message: messageData,
+              });
+            }
+          });
+          pendingMessages = []; // free memory
+
+          // build map of userProfileName to primaryKeys
+          // if we have verified primaryKeys/the claimed relation
+          if (verifiedPrimaryPKs.length) {
+            // get final list of verified chat server profile names
+            const verifiedDeviceResults = await this.serverAPI.getUsers(
+              verifiedPrimaryPKs
+            );
+            // console.log('verifiedDeviceResults', verifiedDeviceResults)
+
+            // go through verifiedDeviceResults
+            const newPrimaryUserProfileName = {};
+            verifiedDeviceResults.forEach(user => {
+              newPrimaryUserProfileName[user.username] = user.name;
+            });
+            // replace whole
+            this.primaryUserProfileName = newPrimaryUserProfileName;
+          }
+
+          // process remaining messages
+          const ourNumber = textsecure.storage.user.getNumber();
+          Object.keys(slaveMessages).forEach(slaveKey => {
+            // prevent our own sent messages from coming back in
+            if (slaveKey === ourNumber) {
+              // we originally sent these
+              return;
+            }
+            const primaryPubKey = slavePrimaryMap[slaveKey];
+            slaveMessages[slaveKey].forEach(messageDataP => {
+              const messageData = messageDataP; // for linter
+              if (slavePrimaryMap[messageData.source]) {
+                // rewrite source, profile
+                messageData.source = primaryPubKey;
+                messageData.message.profile.displayName = this.primaryUserProfileName[
+                  primaryPubKey
+                ];
+              }
+              this.serverAPI.chatAPI.emit('publicMessage', {
+                message: messageData,
+              });
+            });
+          });
+        } // end if there are pending pubkeys to look up
+
+        // console.log('pendingMessages len', pendingMessages.length);
+        // console.log('pendingMessages', pendingMessages);
+        // find messages for original slave key using slavePrimaryMap
+        if (pendingMessages.length) {
```

Comment:
This looks like it is just an else for the above if?

Reply:
Also I'm not really understanding what this part is doing?
Since the above if `pubkeys.length` always resets pendingMessages to [], this guard is essentially an else
But from the logic above, if you have any pending messages, you *must* have at least one pubkey as well
So I don't see how this could ever be entered?

Reply:
There was probably a time it made sense but after the revisions, refactoring and after viewing it with fresh eyes, I've come the same conclusions you've drawn above: not only an else but one that should never be hit.